Source/Extensions/Blazorise.TreeView/TreeView.razor

```diff
@@ -0,0 +1,39 @@
+﻿@typeparam TNode
+@inherits TreeViewBase<TNode>
+<div class="tree-view @(Visible ? String.Empty : "tree-view-hidden")">
+
+    @foreach ( TNode node in Nodes ?? Enumerable.Empty<TNode>() )
+    {
+        bool nodeExpanded = ExpandedNodes.Contains( node );
+        bool nodeSelected = node.Equals( SelectedNode );
+        bool hasChildren = HasChildNodes( node );
+
+        <div>
+            @if ( hasChildren == true )
+            {
+                <span class="tree-view-icon" @onclick="@(() => OnToggleNode(node, !nodeExpanded))">
+                    <i class="@(nodeExpanded ? CollapseNodeIconClass : ExpandNodeIconClass)"></i>
+                </span>
+            }
+
+            <div class="tree-view-title">
+                <span class="@NodeTitleClass @(nodeSelected ? NodeTitleSelectedClass : String.Empty)" @onclick="@(() => OnSelectNode(node))">
+                    @TextContent( node )
+                </span>
+            </div>
+
+            @if ( hasChildren == true )
+            {
+                <TreeView Nodes="SetChildNodes(node)"
+                          TextContent="TextContent"
+                          SetChildNodes="SetChildNodes"
+                          ExpandedNodes="ExpandedNodes"
+                          ExpandedNodesChanged="ExpandedNodesChanged"
+                          SelectedNode="SelectedNode"
+                          SelectedNodeChanged="SelectedNodeChanged"
+                          Visible="nodeExpanded"
+                          HasChildNodes="HasChildNodes" />
```

Comment:
Can `HasChildNodes` be detected from data without specifying it explicitly?

Reply:
Unfortunately not (at least I could not separate them), The TreeView is (probably) unique in that it is recursively created within itself, and the data class that is provided (TNode) can be anything, the only way I know of passing the child nodes, visibility and whether the child nodes themselves contain child nodes is by referencing them in the component. Really the users would probably never set the visibility of a treeview themselves, but I still need a Visibility parameter hooked up to nodeExpanded (so the child treeviews display when the node is expanded).

Reply:
Maybe one way is to have internal component named `_TreeView` that will have those parameters and have `TreeView` as a public component that is a wrapper around `_TreeView`.

Similar concept I have for DataGrid. Also that way we can have CascadingParameter(s) like SelectedNode.